include/swift/SIL/OwnershipUtils.h

```diff
@@ -665,25 +663,26 @@ struct InteriorPointerOperand {
     llvm_unreachable("Covered switch isn't covered?!");
   }
 
-  /// Compute the list of implicit uses that this interior pointer operand puts
-  /// on its parent guaranted value.
+  /// Transitively compute the list of uses that this interior pointer operand
+  /// puts on its parent guaranted value.
   ///
   /// Example: Uses of a ref_element_addr can not occur outside of the lifetime
   /// of the instruction's operand. The uses of that address act as liveness
   /// requirements to ensure that the underlying class is alive at all use
   /// points.
-  bool getImplicitUses(SmallVectorImpl<Operand *> &foundUses,
-                       std::function<void(Operand *)> *onError = nullptr) {
-    return getImplicitUsesForAddress(getProjectedAddress(), foundUses, onError);
+  bool findTransitiveUses(SmallVectorImpl<Operand *> &foundUses,
+                          std::function<void(Operand *)> *onError = nullptr) {
+    return findTransitiveUsesForAddress(getProjectedAddress(), foundUses,
+                                        onError);
   }
 
   /// The algorithm that is used to determine what the verifier will consider to
-  /// be implicit uses of the given address. Used to implement \see
-  /// getImplicitUses.
+  /// be transitive uses of the given address. Used to implement \see
+  /// findTransitiveUses.
   static bool
-  getImplicitUsesForAddress(SILValue address,
-                            SmallVectorImpl<Operand *> &foundUses,
-                            std::function<void(Operand *)> *onError = nullptr);
+  findTransitiveUsesForAddress(SILValue address,
```

Comment:
Eventually we are going to want to represent convert_escape_to_no_escape as an interior pointer and its result is not an address. I am not sure what the appropriate name is though. Maybe findTransitiveProtectedUses? With the idea that we are finding the uses that the borrow scope is ensuring are valid... open to better ideas.

Reply:
Now that I am reading in a little more detail, maybe we would create a separate findTransitiveUsesForObject for the convert_escape_to_no_escape case.

Reply:
I wanted to revisit these helpers but didn't want to put the time in on it for this PR